core/src/test/java/io/aiven/klaw/service/TopicControllerServiceTest.java

```diff
@@ -382,74 +685,219 @@ public void getAllTopics() {
   }
 
   @Test
-  @Order(16)
-  public void getTopicsSuccess1() throws Exception {
+  @Order(29)
+  public void getAllTopics2() {
+    stubUserInfo();
+
+    when(commonUtilsService.getFilteredTopicsForTenant(any())).thenReturn(utilMethods.getTopics());
+    when(handleDbRequests.getSyncTopics(any(), any(), anyInt()))
+        .thenReturn(utilMethods.getTopics());
+    when(commonUtilsService.getTeamId(anyString())).thenReturn(3);
+
+    List<String> result = topicControllerService.getAllTopics(true);
+    assertThat(result).hasSize(1);
+    assertThat(result.get(0)).isEqualTo("testtopic");
+  }
+
+  @Test
+  @Order(30)
+  public void saveTopicDocumentation() throws KlawException {
+    stubUserInfo();
+    TopicInfo topicInfo = utilMethods.getTopicInfo();
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTopicTeam(anyString(), anyInt()))
+        .thenReturn(List.of(getTopic("testtopic")));
+    when(commonUtilsService.getFilteredTopicsForTenant(any())).thenReturn(utilMethods.getTopics());
+    when(commonUtilsService.getTeamId(anyString())).thenReturn(3);
+
+    when(handleDbRequests.updateTopicDocumentation(any()))
+        .thenReturn(ApiResultStatus.SUCCESS.value);
+
+    ApiResponse apiResponse = topicControllerService.saveTopicDocumentation(topicInfo);
+    assertThat(apiResponse.getResult()).isEqualTo(ApiResultStatus.SUCCESS.value);
+  }
+
+  @Test
+  @Order(31)
+  public void saveTopicDocumentationFailure() throws KlawException {
+    stubUserInfo();
+    TopicInfo topicInfo = utilMethods.getTopicInfo();
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTopicTeam(anyString(), anyInt()))
+        .thenReturn(List.of(getTopic("testtopic")));
+    when(commonUtilsService.getFilteredTopicsForTenant(any())).thenReturn(utilMethods.getTopics());
+    when(commonUtilsService.getTeamId(anyString())).thenReturn(1);
+
+    when(handleDbRequests.updateTopicDocumentation(any()))
+        .thenReturn(ApiResultStatus.FAILURE.value);
+
+    ApiResponse apiResponse = topicControllerService.saveTopicDocumentation(topicInfo);
+    assertThat(apiResponse.getResult()).isEqualTo(ApiResultStatus.FAILURE.value);
+  }
+
+  @Test
+  @Order(32)
+  public void getTopicDetailsPerEnvFailure1() {
+    stubUserInfo();
+    String envId = "1", topicName = "testtopic";
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTopics(anyString(), anyInt())).thenReturn(Collections.emptyList());
+    when(commonUtilsService.getEnvsFromUserId(anyString()))
+        .thenReturn(new HashSet<>(Collections.singletonList("1")));
+
+    Map<String, Object> topicDetailsPerEnvResponse =
+        topicControllerService.getTopicDetailsPerEnv(envId, topicName);
+    assertThat(topicDetailsPerEnvResponse.get("error")).isEqualTo("Topic does not exist.");
+  }
+
+  @Test
+  @Order(33)
+  public void getTopicDetailsPerEnvFailure2() {
```

Comment:
We should give it a proper name instead of `1` and `2`.

Reply:
Agree, updated.